app/services/page_cloner.rb

```diff
@@ -0,0 +1,79 @@
+class PageCloner
+  def self.clone!(page)
+    new(page).clone!
+  end
+
+  def initialize(page)
+    @page = page
+  end
+
+  def clone!
+    cloned_page = @page.dup
+
+    # TODO: I'm thinking to have the user of this API
+    # decide what the new title should be and have it
+    # passed in to clone! It's not this class' responsibility
+    # to do the necessary checks against AK
```

Comment:
AK check will be orchestrated at the controller level - we may even already have a class for title checking against AK.